src/hera/validators.py

```diff
@@ -21,8 +21,10 @@ def validate_name(name: str, max_length: Optional[int] = None) -> str:
     if "_" in name:
         raise ValueError("Name cannot include an underscore")
 
-    pattern = r"[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
-    match_obj = re.match(pattern, name)
+    pattern = r"[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
```

Comment:
Small: any chance [this link](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/) can be included as a comment or a `Notes` in the docstring? I _think_ that documents the motivation for this specific validation and it would be nice to have a record in code that points contributors to it

Reply:
Added `Notes` in the docstring.